File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/web/RecipeModifyRequest.java
package com.tarbadev.witchcraft.web;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RecipeModifyRequest {
  private Integer id;
  private String url;
  private String name;
  private String imgUrl;
  @Builder.Default
  private Boolean favorite = false;
  private List<IngredientModifyRequest> ingredients;
  private List<StepModifyRequest> steps;
}

File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/recipes/rest/RecipeModifyRequest.kt
package com.tarbadev.witchcraft.recipes.rest

data class RecipeModifyRequest(
    val id: Int = 0,
    val url: String = "",
    val name: String = "",
    val imgUrl: String = "",
    val favorite: Boolean = false,
    val ingredients: List<IngredientModifyRequest> = emptyList(),
    val steps: List<StepModifyRequest> = emptyList()
)
